include/oneapi/dpl/internal/distributed_ranges_impl/detail/iterator_adaptor.hpp

```diff
@@ -23,17 +23,12 @@
 namespace oneapi::dpl::experimental::dr
 {
 
-namespace
-{
-
 template <typename R>
 concept has_segments_method = requires(R r)
 {
     {r.segments()};
 };
 
-} // namespace
-
```

Comment:
This change adds `concept has_segments_method` into `namespace oneapi::dpl::experimental::dr`, which is sort of equivalent to making it a public API. Is it intentional? If not, I would use `namespace __detail` instead.

Just in case, in my comment https://github.com/oneapi-src/oneDPL/pull/1479#discussion_r1686830006 about unnamed namespaces, I did not imply it is always bad - I pointed out to a particular strange case of an unnamed namespace used directly within the default global namespace.

Reply:
Should be in `__detail`.
`__detail::has_segments_method` concept is duplicated in `iterator_adaptor.hpp` -> to be removed